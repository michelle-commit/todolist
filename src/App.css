.container input{
  margin-left: 40%;
  margin-top: 2%;
  width: 20%;
  height: 20px;
  border: 2px solid black;
  border-radius: 5px;
}
.azerty{
  display: flex;
  flex-direction: row;
  margin-top: 5%;
}
.todo{
  flex-basis: 50%;

}
.todo h3{
  text-align: center;
  align-items: center;
}
.done{
  flex-basis: 50%;
}
.done h3{
  text-align: center;
  align-items: center;
}