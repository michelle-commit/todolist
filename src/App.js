import logo from './logo.svg';
import './App.css';
import React, { useState } from 'react';

function TodoList() {
    const [todo, setTodo] = useState('');
    const [doing, setDoing] = useState([]);
   const [done, setDone] = useState([]);

    const handleSubmit = (e) => {
    e.preventDefault();
    setDoing([...doing, todo]);
    setTodo('');
  }

  const handleDelete = (index) => {
    const newDoing = [...doing];
    newDoing.splice(index, 1);
    setDoing(newDoing);
  }

   const handleCheck = (index) => {
    const currentTodo = doing[index];
    handleDelete(index);
    setDone([...done, currentTodo]);
  }
  return (
    <div className="App">
      <div className='container'>
        <form onSubmit={handleSubmit} data-testid = "input-1">
          <input 
            type="text" 
            value={todo} 
            onChange={e => setTodo(e.target.value)} onKeyDown={(e)=>{e.key === 'Enter' && handleSubmit()}}
            placeholder="Add task"
          />
        </form>
      <div className='azerty'>
        <div className='todo'>
          <h3>TODO</h3>
          <ul>
            {doing.map((todo, index) => (
              <li key={index}>
                {todo}
                <input 
                  type="checkbox" 
                  onClick={() => handleCheck(index)} 
                />
              </li>
            ))}
          </ul>
        </div>
        <div className='done'>
          <h3>DONE</h3>
          <ul>
            {done.map((todo, index) => (
              <li key={index}>{todo}</li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  </div>
  );
}

export default TodoList;


