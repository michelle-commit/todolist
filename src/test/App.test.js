import { fireEvent, getByTestId, render } from '@testing-library/react';
import App from '../App';
import renderer from 'react-test-renderer';

describe('nnnnnn',()=>{
    it(' App content something',()=>{
        const {getByText}=render(<App/>);
        expect (getByText('TODO')).toBeInTheDocument();
    })
    it('shows the expected placeholder text',()=>{
        const { getByPlaceholderText } = render(<App/>);
        const input = getByPlaceholderText(/Add task/);
        expect(input).toBeInTheDocument();
       
    }) 
    it('input values changes ', ()=>{
        const { getByPlaceholderText } = render(<App/>);
        const input = getByPlaceholderText('Add task');
        fireEvent.change(input, { target: { value: 'test' } });
        expect(input.value).toBe('test');
    })  
    it("should call handleSubmit",()=>{
        const todo = jest.fn();
        const {getByTestId} = render(<App handleSubmit={todo()}/>)
        fireEvent.keyPress( getByTestId('input-1') , {key:'Enter', code: 13})
        expect(todo).toHaveBeenCalled();
    })    
    it("input form submit the task",()=>{
        const{ getByPlaceholderText , getByTestId}= render(<App/>);
        const input = getByPlaceholderText('Add task');
        const form = getByTestId('input-1');
        fireEvent.change(input, {target : { value : 'test'}});
        fireEvent.submit(form);
        const { doing } = render(<App />).container.state;
        expect(doing).toContain('test')

    })

    it('matches snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
      });
});







