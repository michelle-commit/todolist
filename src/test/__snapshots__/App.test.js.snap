// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nnnnnn matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <form
      data-testid="input-1"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Add task"
        type="text"
        value=""
      />
    </form>
    <div
      className="azerty"
    >
      <div
        className="todo"
      >
        <h3>
          TODO
        </h3>
        <ul />
      </div>
      <div
        className="done"
      >
        <h3>
          DONE
        </h3>
        <ul />
      </div>
    </div>
  </div>
</div>
`;
